malicious-site-protection/malicious-site-protection-impl/src/main/kotlin/com/duckduckgo/malicioussiteprotection/impl/data/MaliciousSiteRepository.kt

```
@@ -0,0 +1,148 @@
+
+            (maliciousSiteDao.getLatestRevision()?.filter { it.type == Type.FILTER_SET.name } ?: listOf()).let { latestRevision ->
+                val phishingFilterSetRevision = latestRevision.firstOrNull() { it.feed == PHISHING.name }?.revision ?: 0
+                val phishingFilterSet: FilterSetResponse? = if (networkRevision > phishingFilterSetRevision) {
+                    maliciousSiteService.getPhishingFilterSet(phishingFilterSetRevision)
+                } else {
+                    null
+                }
+                val malwareFilterSetRevision = latestRevision.firstOrNull() { it.feed == MALWARE.name }?.revision ?: 0
+                val malwareFilterSet: FilterSetResponse? = if (networkRevision > malwareFilterSetRevision) {
+                    maliciousSiteService.getMalwareFilterSet(malwareFilterSetRevision)
+                } else {
+                    null
+                }
+
+                maliciousSiteDao.updateFilters(
+                    phishingFilterSet?.let {
+                        PhishingFilterSetWithRevision(
+                            it.insert.map { insert -> Filter(insert.hash, insert.regex) }.toSet(),
+                            it.delete.map { delete -> Filter(delete.hash, delete.regex) }.toSet(),
+                            it.revision,
+                            it.replace,
+                        )
+                    },
+                )
+                maliciousSiteDao.updateFilters(
+                    malwareFilterSet?.let {
+                        MalwareFilterSetWithRevision(
+                            it.insert.map { insert -> Filter(insert.hash, insert.regex) }.toSet(),
+                            it.delete.map { delete -> Filter(delete.hash, delete.regex) }.toSet(),
+                            it.revision,
+                            it.replace,
+                        )
+                    },
+                )
+            }
+        } catch (e: Exception) {
```

if we try-catch, we will never retry (in the worker) following the backoff criteria. I believe that should be ok, but fyi for you to consider if we should let it crash, or maybe just check the error and decide.